_source/_posts/2022-08-12-ci-cd-circleci-spinnaker-microservices.md

```diff
@@ -0,0 +1,949 @@
+---
+layout: blog_post
+title: "CI/CD with CircleCI and Spinnaker for JHipster Microservices"
+author: jimena-garbarino
+by: contractor
+communities: [devops,security,java]
+description: "Learn how to set up continuous integration with CircleCI and continuous delivery with Spinnaker for a JHipster microservices architecture."
+tags: [java, ci, cd, jhipster, circleci, spinnaker]
+tweets:
+- "CI/CD are essential practices for modern software development. Learn how to test and deploy microservices with @circleci and @spinnakerio in this awesome tutorial!"
+- "We love this tutorial from @indiepopart! It shows how to create Spring Boot microservices, CI them with CircleCI, and deploy them with Spinnaker to Google Cloud."
+- "Generate microservices and @kubernetesio configuration with @jhipster, then automate all its CI/CD! This tutorial shows you how!"
+image: blog/jhipster-ci-cd/ci-cd-baby.jpg
+type: conversion
+---
+
+Continuous integration and delivery (CI/CD) are essential practices for modern software development. In this post we cover the basics of how to add CI/CD for a JHipster microservices architecture and Kubernetes as the target cloud deployment environment.
+
+Briefly, continuous integration is the practice of integrating code into the main branch of a shared repository early and often. Instead of integrating features at the end of a development cycle, code is integrated with the shared repository multiple times throughout the day. Each commit triggers automated tests, so issues are detected and fixed earlier and faster, improving team confidence and productivity. The chosen continuous integration platform is from CircleCI, a company founded in 2011 and headquartered in San Francisco. They offer a free cloud to test their services.
```

Comment:
```suggestion
Briefly, *continuous integration* is the practice of integrating code into the main branch of a shared repository early and often. Instead of integrating features at the end of a development cycle, code is integrated with the shared repository multiple times throughout the day. Each commit triggers automated tests, so issues are detected and fixed earlier and faster, improving team confidence and productivity. The chosen continuous integration platform is from CircleCI, a company founded in 2011 and headquartered in San Francisco. They offer a free cloud to test their services.
```
for scannability